lib/index.js

```diff
@@ -19,23 +31,47 @@ module.exports = (referenceDir, config, { watch, fingerprint, sourcemaps, compac
 		fingerprint,
 		exitOnError: !watch
 	});
-	let watcher = watch && makeWatcher(config.watchDirs, referenceDir,
-			assetManager.resolvePath);
 	let browsers = browserslist.findConfig(referenceDir) || {};
 
 	let plugins = Object.assign({}, DEFAULTS.plugins, config.plugins);
+	let buckets = {
+		static: [],
+		scripts: [],
+		styles: [],
+		markup: []
+	};
 	Object.keys(plugins).forEach(type => {
 		let pluginConfig = config[type];
 		if(!pluginConfig) {
 			return;
 		}
 
-		let plugin = plugins[type];
+		let bucket = plugins[type].bucket;
+		let plugin = plugins[type].package;
```

Comment:
The name `package` is not great as it is a reserved keyword and can't be used as a variable name. I really think that it's a good name for this. Any other suggestions?

Reply:
I know it's somewhat awkardly redundant because of the parent object, but instead of "package" I'd recommend "plugin"; that's pretty much what's expected there (either as a package reference or directly as a function)?

PS: Destructuring, please?